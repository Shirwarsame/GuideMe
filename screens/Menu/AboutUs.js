import React from "react";
import { View, Text, StyleSheet, SafeAreaView, SectionList } from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import { Paragraph } from 'react-native-paper';
import { Icon, Button } from "native-base";
import { Feather } from "@expo/vector-icons";

export const renderSeparator = () => {
    return (
        <View
            style={{
                height: 1,
                width: "100%",
                backgroundColor: "#353A50",
            }}
        />
    );
};

function AboutUs(props) {

    const goToMenu = () => {
        props.navigation.openDrawer();
    };

    const data = [{
        title: "Team - Github",
        data: ["Mohanad Arafe - mohanadarafe",
            "Christophe Shaka Bahenduzi - csbduzi",
            "Kenza Boulisfane - bkenza",
            "Jérémie Chouteau - JayChoteau",
            "Baddredine Loulidi - badred123",
            "Firas Sawan - firassawan",
            "Alain Job Uzarama Uwe - alainjobs",
            "Shirwa Warsame - shirwarsame",
            "Hussain Witwit - hussainwitwit"
        ]
    }]

    return (
        <View style={styles.container}>
            <View style={styles.menuButtonContainer}>
                <Button transparent style={styles.menuButton} onPress={goToMenu}>
                    <Feather name="menu" style={styles.icon} />
                </Button>
            </View>

            <Text style={styles.mainLabel}>About Us</Text>
            <Text style={styles.aboutLabel}>Our App</Text>

            <View style={styles.scrollViewTextContainer}>
            <ScrollView style={{flex:1}}>
                 <Paragraph style={styles.aboutText}>GuideMe is a software application made for commuters of Concordia University. The purpose of the application is to serve users a map for both the Loyola & Sir George Williams Campuses. The application gives a new refined way to navigate on campus offering route recommendations & various methods of travelling.
                  </Paragraph>
                 </ScrollView>
            </View>
           
            <Text style={styles.teamLabel}>Our Team</Text>

            <SafeAreaView style={styles.scrollTextContainer}>
                <SectionList
                    sections={data}
                    renderItem={({ item }) => <Text style={styles.listItem}>{item}</Text>}
                    renderSectionHeader={({ section }) => <Text style={styles.sectionHeader}>{section.title}</Text>}
                    keyExtractor={(item, index) => index}
                    ItemSeparatorComponent={renderSeparator}
                />
            </SafeAreaView>


        </View >
    );
}

export const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        justifyContent: "space-between",
        height: "100%",
        width: "100%",
        backgroundColor: "#2A2E43",
        flex:1
    },
    scrollViewTextContainer:{
        bottom: "57%",
        height: 150,
        width: 415,
    },
    mainLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 25,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "15%"
    },
    aboutLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 18,
        fontWeight: "bold",
        fontFamily: "encodeSansExpanded",
        top: "22%"
    },
    aboutText: {
        position: "absolute",
        alignSelf: "center",
        fontSize: 12,
        fontFamily: "encodeSansExpanded",
        color: "#D3D3D3",
        top: "27%",
        width: "85%",
        textAlign: "center"
    },
    teamLabel: {
        color: "#FFFFFF",
        position: "absolute",
        fontSize: 18,
        fontFamily: "encodeSansExpanded",
        top:"43%"
    },
    icon: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
        left: "6%",
        color: "#FFFFFF",
        fontSize: 35
    },
    menuButton: {
        height: "100%",
        width: "100%",
        flexDirection: "row",
    },
    menuButtonContainer: {
        width: "100%",
        height: "6%",
        flexDirection: "column",
        justifyContent: "space-around",
        alignContent: "center",
        alignItems: "center",
        top: "7%"
    },

    scrollTextContainer: {
        width: "100%",
        height: "52%",
        bottom: "0%",
        position: "absolute",
    },
    sectionHeader: {
        paddingTop: 2,
        paddingLeft: 22,
        paddingRight: 10,
        paddingBottom: 2,
        fontSize: 18,
        fontWeight: "bold",
        color: "#ffffff",
        backgroundColor: "#353A50",
        fontFamily: "encodeSansExpanded"
    },
    listItem: {
        padding: 10,
        fontSize: 12,
        height: 44,
        paddingLeft: 22,
        fontFamily: "encodeSansExpanded",
        color: "#D3D3D3",
    },

});

export default AboutUs;
