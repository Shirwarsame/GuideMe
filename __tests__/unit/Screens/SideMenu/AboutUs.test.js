import React from "react";
import AboutUs from "../../../../screens/Menu/AboutUs";
import renderer from "react-test-renderer";
import { TouchableOpacity } from "react-native";
import { shallow } from "enzyme";

describe("About Us component", () => {
    test("renders correctly", () => {
        const tree = renderer.create(<AboutUs />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("renders side menu", () => {
        const navigation = { openDrawer: () => {
            return true;
        }};
        const wrapper = shallow(<AboutUs navigation={navigation} />);
        wrapper.find(TouchableOpacity).props().onPress();
    });
});