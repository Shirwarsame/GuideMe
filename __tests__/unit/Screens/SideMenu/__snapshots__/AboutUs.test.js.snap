// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2A2E43",
      "height": "100%",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": "6%",
        "top": "7%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "20%",
        }
      }
    >
      <Text />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "encodeSansExpanded",
        "fontSize": 25,
        "fontWeight": "bold",
        "position": "absolute",
        "top": "15%",
      }
    }
  >
    About Us
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "encodeSansExpanded",
        "fontSize": 18,
        "fontWeight": "bold",
        "position": "absolute",
        "top": "22%",
      }
    }
  >
    Our App
  </Text>
  <View
    style={
      Object {
        "bottom": "59%",
        "height": "15%",
        "width": "95%",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              "GuideMe is a software application made for commuters of Concordia University. The purpose of the application is to serve users a map for both the Loyola & Sir George Williams Campuses. The application gives a new refined way to navigate on campus offering route recommendations & various methods of travelling.",
            ],
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      sections={
        Array [
          Object {
            "data": Array [
              "GuideMe is a software application made for commuters of Concordia University. The purpose of the application is to serve users a map for both the Loyola & Sir George Williams Campuses. The application gives a new refined way to navigate on campus offering route recommendations & various methods of travelling.",
            ],
          },
        ]
      }
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      stickySectionHeadersEnabled={true}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        />
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontFamily": "encodeSansExpanded",
                "fontSize": 14,
                "lineHeight": 30,
                "padding": 10,
                "textAlign": "center",
              }
            }
          >
            GuideMe is a software application made for commuters of Concordia University. The purpose of the application is to serve users a map for both the Loyola & Sir George Williams Campuses. The application gives a new refined way to navigate on campus offering route recommendations & various methods of travelling.
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
  </View>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "encodeSansExpanded",
        "fontSize": 18,
        "position": "absolute",
        "top": "43%",
      }
    }
  >
    Our Team
  </Text>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "bottom": "0%",
        "height": "52%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              "Mohanad Arafe - mohanadarafe",
              "Christophe Shaka Bahenduzi - csbduzi",
              "Kenza Boulisfane - bkenza",
              "Jérémie Chouteau - JayChoteau",
              "Baddredine Loulidi - badred123",
              "Firas Sawan - firassawan",
              "Alain Job Uzarama Uwe - alainjobs",
              "Shirwa Warsame - shirwarsame",
              "Hussain Witwit - hussainwitwit",
            ],
            "title": "Team - Github",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      renderSectionHeader={[Function]}
      scrollEventThrottle={50}
      sections={
        Array [
          Object {
            "data": Array [
              "Mohanad Arafe - mohanadarafe",
              "Christophe Shaka Bahenduzi - csbduzi",
              "Kenza Boulisfane - bkenza",
              "Jérémie Chouteau - JayChoteau",
              "Baddredine Loulidi - badred123",
              "Firas Sawan - firassawan",
              "Alain Job Uzarama Uwe - alainjobs",
              "Shirwa Warsame - shirwarsame",
              "Hussain Witwit - hussainwitwit",
            ],
            "title": "Team - Github",
          },
        ]
      }
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      stickySectionHeadersEnabled={true}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              Object {
                "backgroundColor": "#353A50",
                "color": "#ffffff",
                "fontFamily": "encodeSansExpanded",
                "fontSize": 18,
                "fontWeight": "bold",
                "paddingBottom": 2,
                "paddingLeft": 22,
                "paddingRight": 10,
                "paddingTop": 2,
              }
            }
          >
            Team - Github
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "encodeSansExpanded",
                  "fontSize": 12,
                  "height": 44,
                  "padding": 10,
                  "paddingLeft": 22,
                }
              }
            >
              Mohanad Arafe - mohanadarafe
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#353A50",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "encodeSansExpanded",
                  "fontSize": 12,
                  "height": 44,
                  "padding": 10,
                  "paddingLeft": 22,
                }
              }
            >
              Christophe Shaka Bahenduzi - csbduzi
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#353A50",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "encodeSansExpanded",
                  "fontSize": 12,
                  "height": 44,
                  "padding": 10,
                  "paddingLeft": 22,
                }
              }
            >
              Kenza Boulisfane - bkenza
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#353A50",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "encodeSansExpanded",
                  "fontSize": 12,
                  "height": 44,
                  "padding": 10,
                  "paddingLeft": 22,
                }
              }
            >
              Jérémie Chouteau - JayChoteau
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#353A50",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "encodeSansExpanded",
                  "fontSize": 12,
                  "height": 44,
                  "padding": 10,
                  "paddingLeft": 22,
                }
              }
            >
              Baddredine Loulidi - badred123
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#353A50",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "encodeSansExpanded",
                  "fontSize": 12,
                  "height": 44,
                  "padding": 10,
                  "paddingLeft": 22,
                }
              }
            >
              Firas Sawan - firassawan
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#353A50",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "encodeSansExpanded",
                  "fontSize": 12,
                  "height": 44,
                  "padding": 10,
                  "paddingLeft": 22,
                }
              }
            >
              Alain Job Uzarama Uwe - alainjobs
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#353A50",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontFamily": "encodeSansExpanded",
                  "fontSize": 12,
                  "height": 44,
                  "padding": 10,
                  "paddingLeft": 22,
                }
              }
            >
              Shirwa Warsame - shirwarsame
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#353A50",
                  "height": 1,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontFamily": "encodeSansExpanded",
                "fontSize": 12,
                "height": 44,
                "padding": 10,
                "paddingLeft": 22,
              }
            }
          >
            Hussain Witwit - hussainwitwit
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
